### ISSUE TITLE
Moving 3dJS project to a new bigger project

### ISSUE THREAD

**Issue by XorTroll** [XorTroll/3dJS#1]
I've seen that the idea of enabling JS hmebrew on a 3DS system is way great.
Basically, I'm developing a new project which consists more or less on this, but not just an interpreter, also with an API and the possibility to create an entire ROM that runs JS code instead of having to interpretate it.
**Brew.js** is the new project's name, and it will have some sub-projects: NTR.js for Nintendo DS, CTR.js for Nintendo 3DS, RVL.js for Wii, HAC.js for Nintendo Switch...
Here you have the link of the organisation with all the repositories: [https://github.com/BrewJS](url)
Hope you enjoy it!

**Comment by RedDuckss** [comment]
Might I suggest changing to WUP.js for the WiiU and NX.js for the Switch? Those would make more sense for those consoles, seeing as internally Nintendo refers to the WiiU by WUP and NX was the codename for the Switch

**Comment by XorTroll** [comment]
Well, the real codename for the Switch is HAC (AFAIK), so I don't think I'm changing it. It's similar with the Wii U.
On the Wii U, the codename is WUP, while the internal name used by devs is Cafe.
On Switch the codename is HAC, while the internal name is NX.
The official SDKs of Nintendo use NX/Cafe, while on retail products HAC/WUP are used.
I understand why are you saying that, but AFAIK WUP and HAC are the real codenames.

**Comment by RedDuckss** [comment]
```
Well, the real codename for the Switch is HAC
```
```
On Switch the codename is HAC, while the internal name is NX
```
Can I get some source on this? Every source I can find about the Switch's codename comes back as NX. I can't find a single resource where it's called HAC

```
On the Wii U, the codename is WUP, while the internal name used by devs is Cafe
```
This is backwards, actually. The WiiU was codenamed "Project Cafe", and Cafe was used as the name of the WiiU's OS. But internally, the WiiU is refereed to by WUP (this can be verified by checking things such as NUS and/or it's corresponding CDN. In NUS Nintendo refers to the 3DS as CTR, the N3DS as KTR (as well as the 2DS, I think?), and WiiU as WUP)

```
The official SDKs of Nintendo use NX/Cafe, while on retail products HAC/WUP are used.
```
I can't speak on the Switch SDK, since I don't have access to it (I know ryanrocks though, and he has access to it (he's the guy who leaked it) so I guess I could ask him), but I do have access to the WiiU's SDK and from what I've seen it never actually refers to the console by it's codename.

**Comment by XorTroll** [comment]
Well, I've accessed both SDKs.
Basically the Wii U Sdk is called Cafe SDK, and all the files from the Switch SDK are called "NX-..."

**Comment by XorTroll** [comment]
https://gbatemp.net/attachments/in-switch-un-sdk-de-la-nintendo-switch-leake-1-jpg.91642/

**Comment by RedDuckss** [comment]
It’s called the “Cafe SDK” because the OS on the console is named “Cafe”. Again, internally Nintendo refers to the WiiU by WUP and NX was the codename of the Switch. I’ve alredy told you where you can verify by that, by checking things NUS, where Nintendo clearly refers to the WiiU by WUP, which is consistent with how they refer to other consoles like the 3DS (on NUS they refer to the 3DS as CTR, N3DS as KTR and WiiU as WUP, all of which match how you’re naming things except for the WiiU)

**Comment by XorTroll** [comment]
That's true, the Wii U is referred as WUP, but if you check NUS you can se that Switch is always called as HAC. BTW, I must tell you that I'm adding support for the followinh platforms: NDS-NTR, 3DS-CTR, GCN-DOL, Wii-RVL, Switch-HAC
See projects here: https://github.com/BrewJS

**Comment by XorTroll** [comment]
It's true that many people won't understand what HAC means, but I would prefer to have the real codenames (those who use hactool executable will know)
I'm not adding Wii U support for the moment, but I'd call it WUP if I added it.

**Comment by RedDuckss** [comment]
```
It's true that many people won't understand what HAC means
```
Seems kinda like bad UX to do this, purposefully naming parts of your codebase in ways that users won't understand but expected to use.

```
I would prefer to have the real codenames
```
The real codename for the Switch is NX though..? Again, NX and HAC are two different things. HAC is what the Switch is referred to internally (the consoles model abbreviation). The codename was NX. If you want to use codenames, then for the Wii you should name it `Revolution.js`, or `Nitro.js` for the DS. What you're using right now are the internal names, which are the model abbreviations, of the consoles. Not the codenames.

```
(those who use hactool executable will know)
```
I've used hactool and in no way did it make any mention to HAC being related to the Switch besides the tools name. I had no idea there was even a relation until now, so I find it doubtful that many others will based solely on the use of this tool.

```
I'm not adding Wii U support for the moment
```
I'm aware of this, if you go back and read my initial comment I only brought up the WiiU because I misread "Wii" as "WiiU". I even edited the comment to reflect the mistake.

**Comment by XorTroll** [comment]
Yep, I'm a bit messed up with the concepts of code names or internal names. I've finally decided to call the repositories just like
 platforms' names.
BTW, I think I'm releasing the Switch interpreter tomorrow, or maybe in a few days. Libnx doesn't have support for native Switch keyboard for the moment, so I'm using a different way to interpretate code.
